tests/monitoring_test.go

```diff
@@ -271,22 +271,27 @@ var _ = Describe("Prometheus Alerts", func() {
 			return vmName
 		}
 
-		It("[test_id:TODO] Should not create kubevirt_ssp_vm_rbd_volume when not using DataVolume or PVC", func() {
-			vmName := createResources(false, true)
-			seriesShouldNotBeDetected(fmt.Sprintf("kubevirt_ssp_vm_rbd_volume{name='%s'}", vmName))
-			alertShouldNotBeActive("VirtualMachineCRCErrors")
-		})
-
 		It("[test_id:TODO] Should not fire VirtualMachineCRCErrors when rxbounce is enabled", func() {
 			vmName := createResources(true, true)
-			waitForSeriesToBeDetected(fmt.Sprintf("kubevirt_ssp_vm_rbd_volume{name='%s', rxbounce_enabled='true'}", vmName))
+			waitForSeriesToBeDetected(fmt.Sprintf("kubevirt_ssp_vm_rbd_block_volume_without_rxbounce{name='%s'}", vmName))
 			alertShouldNotBeActive("VirtualMachineCRCErrors")
 		})
 
 		It("[test_id:10549] Should fire VirtualMachineCRCErrors when rxbounce is disabled", func() {
 			vmName := createResources(true, false)
-			waitForSeriesToBeDetected(fmt.Sprintf("kubevirt_ssp_vm_rbd_volume{name='%s', rxbounce_enabled='false'}", vmName))
+			waitForSeriesToBeDetected(fmt.Sprintf("kubevirt_ssp_vm_rbd_block_volume_without_rxbounce{name='%s'}", vmName))
 			waitForAlertToActivate("VirtualMachineCRCErrors")
+
+			err := apiClient.Delete(ctx, vm)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func(g Gomega) {
+				err := apiClient.Get(ctx, types.NamespacedName{Name: vmName, Namespace: strategy.GetNamespace()}, vm)
+				g.Expect(err).To(HaveOccurred())
+				g.Expect(errors.IsNotFound(err)).To(BeTrue())
```

Comment:
Please use `To(MatchError(...))` instead.

Reply:
I see that `MatchError` is usually used for checking the error contains a given string
Here I think it is better to take advantage of all the checks `errors.IsNotFound` provides

Reply:
`MatchError` is able to take the matcher fn: https://onsi.github.io/gomega/#matcherrorexpected-interface

Reply:
but `k8s.io/apimachinery/pkg/api/errors` does not provide a way to compare to `metav1.StatusReasonNotFound`, its encapsulated in a `APIStatus` struct:

```
if status, ok := err.(APIStatus); ok || errors.As(err, &status) {
	return status.Status().Reason, status.Status().Code
}
```

Reply:
`g.Expect(err).To(MatchError(errors.IsNotFound, "IsNotFound"))`

Does this work?

Reply:
it does :tada: 
updated